### PR TITLE
Enable MySQL unix socket connection

### PR DESCRIPTION
**Pull request by k-alin**
Enables connection to mysql database with unix socket. Location of socket file has to be specified in local.config.php (or local.config.vagrant.php for vm managed by vagrant).

### CODE REVIEW

File: src/Database/Database.php [hunk 1]
```diff
@@ -114,6 +114,7 @@ public function connect()
 		$pass    = trim($this->configCache->get('database', 'password'));
 		$db      = trim($this->configCache->get('database', 'database'));
 		$charset = trim($this->configCache->get('database', 'charset'));
+                $socket  = trim($this->configCache->get('database', 'socket'));
```
Comment (nupplaphil) [inline src/Database/Database.php]:
```suggestion
		$socket  = trim($this->configCache->get('database', 'socket')); 
```

Code Standard - Wrong indent

File: src/Database/Database.php [hunk 2]
```diff
@@ -134,8 +135,12 @@ public function connect()
...
 				$connect .= ";charset=" . $charset;
 			}
-
-			try {
+			
+                        if ($socket) {
+                                $connect .= ";$unix_socket=" . $socket;
+                        }
+			
+                        try {
```
Comment (nupplaphil) [inline src/Database/Database.php]:
```suggestion

		if ($socket) {
			$connect .= ";$unix_socket=" . $socket;
		}

		try {
```

Code standard - wrong indent

File: src/Database/Database.php [hunk 3]
```diff
@@ -134,8 +135,12 @@ public function connect()
 			if ($charset) {
 				$connect .= ";charset=" . $charset;
 			}
-
-			try {
+			
+                        if ($socket) {
+                                $connect .= ";$unix_socket=" . $socket;
```
Comment (nupplaphil) [inline src/Database/Database.php]: could the unix_socket param be in conflict with a port parameter? Should we catch it beforehand?

File: src/Database/Database.php [hunk 4]
```diff
@@ -160,6 +165,11 @@ public function connect()
 				if ($charset) {
 					$this->connection->set_charset($charset);
 				}
+                                
+                                if ($socket) {
+                                        $this->connection->set_socket($socket);
+                                }
+
```
Comment (nupplaphil) [inline src/Database/Database.php]:
```suggestion

				if ($socket) {
					$this->connection->set_socket($socket);
				}

```

Code standard - wrong indent

**Review by nupplaphil — CHANGES_REQUESTED** [review]
Thank you for your contribution!!

Two things I noticed:
- Code standards (indent)
- The question about the discrepancy about port and socket

**Comment by nupplaphil** [comment]
Another thing, please adapt the ENV settings as well:
https://github.com/friendica/friendica/blob/develop/static/env.config.php#L28

And maybe we/you could add this setting to the autoinstaller well (I think this would be possible in a follow up PR as well):
https://github.com/friendica/friendica/blob/develop/src/Console/AutomaticInstallation.php#L81

File: src/Database/Database.php (hunk 3 above)
Comment (k-alin) [reply src/Database/Database.php]: As far as I'm aware they shouldn't conflict? The type of connection is determined by the value of the hostname parameter (more info on that [here](https://www.php.net/manual/en/mysqli.construct.php)). If it's localhost or null, sockets/pipes will be used. It looks like you could provide both parameters, and if you pass an IP address (or localhost / null) to the hostname parameter, the socket (or port) parameter should be ignored.

File: src/Database/Database.php [hunk 5]
```diff
@@ -135,7 +136,11 @@ public function connect()
 				$connect .= ";charset=" . $charset;
 			}
 
-			try {
+		if ($socket) {
+			$connect .= ";$unix_socket=" . $socket;
+		}
+
+		try {
```
Comment (MrPetovan) [inline src/Database/Database.php]: Code standards: Please normalize the indentation level.
```suggestion
			if ($socket) {
				$connect .= ";$unix_socket=" . $socket;
			}

			try {
```

**Review by MrPetovan — CHANGES_REQUESTED** [review]

File: src/Database/Database.php [hunk 6]
```diff
@@ -114,6 +114,7 @@ public function connect()
 		$pass    = trim($this->configCache->get('database', 'password'));
 		$db      = trim($this->configCache->get('database', 'database'));
 		$charset = trim($this->configCache->get('database', 'charset'));
+		$socket  = trim($this->configCache->get('database', 'socket'));
```
Comment (nupplaphil) [inline src/Database/Database.php:117]: Do we need an entry at `static/defaults.config.php` as well? At least for documentation purpose

**Review by MrPetovan — APPROVED** [review]
Thank you for your work!

**Comment by nupplaphil** [comment]
Great change! I think I will use it for the docker images as well :) thank you

**Comment by k-alin** [comment]
Thank you both for your feedback :)

**Pull request merged**